File: matlab/cayley.m
% rotation matrix by Cayley transform
function R = cayley(p)
    T = [0 -p(3) p(2); p(3) 0 -p(1); -p(2) p(1) 0];
    R = (eye(3) - T)/(eye(3) + T);
end

File: python/cayley.py
import numpy as np



# rotation matrix by Cayley transform
def cayley(p):
    id = np.eye(3)
    T = np.array([[0, -p[2], p[1]], [p[2], 0, -p[0]], [-p[1], p[0], 0]])
    R = np.linalg.solve(id - T, id + T)
    return R.T